### PR TITLE
Build workflow updates

### PR DESCRIPTION
**Pull request by iampearceman**
https://github.com/user-attachments/assets/826d9451-fbc5-44e9-a27b-e98fb1e7c2fc

After the initial review, I will modify and add the rest.

### CODE REVIEW

File: mint.json [hunk 1]
```diff
@@ -127,118 +126,139 @@
...
-        "workflow/content",
-        "workflow/digest",
-        "workflow/delay"
+        "architecture/overview",
+        "architecture/diagrams"
       ]
     },
-         
     {
-      "group": "Channels",
```
Comment (scopsy) [inline mint.json:143]: @iampearceman I don't really like this 4 level nesting here :( Let's either flatten it, so each channel is in it's own row and not inside the Step Channels menu, or move it to it's own section like it was before

File: mint.json [hunk 2]
```diff
@@ -127,118 +126,139 @@
...
+              "icon": "message-sms",
               "pages": [
-                "integrations/providers/push/fcm",
-                "integrations/providers/push/apns",
-                "integrations/providers/push/expo-push",
-                "integrations/providers/push/onesignal",
-                "integrations/providers/push/pushpad",
-                "integrations/providers/push/push-webhook",
-                "integrations/providers/push/pusher-beams"
+                "workflow/channels/sms",
```
Comment (scopsy) [inline mint.json:211]: Seems like you changed all the urls for those pages, it will affect SEO. We will need to have redirects to them. But, I don't think the delivery providers should live under workflow, would keep it as parallel concept

File: mint.json [hunk 3]
```diff
@@ -127,118 +126,139 @@
...
       "pages": [
+        "workflow/overview",
         {
-          "group": "In-App",
-          "icon": "bell",
-          "pages": ["integrations/providers/in-app/overview"]
+          "group": "How-To",
+          "pages": [
+            "workflow/how-to/build-a-workflow"
+          ]
```
Comment (scopsy) [inline mint.json:142]: Let's remove this group and just call it Building workflows

File: workflow/overview.mdx [hunk 4]
```diff
@@ -0,0 +1,176 @@
+---
+title: "Workflow Builder Overview"
+sidebarTitle: "Overview"
+---
+
+Our browser-based **Workflow Builder (Editor)** makes it simple to create personalized, automated messaging workflows—no coding required.
+
+## Workflow Elements / Building Blocks
```
Comment (scopsy) [inline workflow/overview.mdx:8]: Let's decide here on one name

File: workflow/overview.mdx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+### Action Step
+
+**Action Steps** are purpose built functions that help manage the workflow flow. **Action Steps** can be executed before and after **Channel Steps**. 
+For example, you can use a **Delay** to pause a workflow for a certain duration before sending a notification.
+
+There are three types of Action Steps supported by Novu:
+
+- Delay
+- Digest
+- Custom
```
Comment (scopsy) [inline workflow/overview.mdx:80]: Let's add here links and in any other relevant place

File: workflow/overview.mdx [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+
+- **Scheduled Notifications**
+  - Automates recurring communication
+  - Supports cron-based scheduling
+  - Easily manage reminders, routine updates
+
+- **Parallel Notification Delivery**
+  - Sends multiple notifications simultaneously
+  - Ensures successful completion of all notifications
+  - Minimizes delays in multi-channel communication
```
Comment (scopsy) [inline workflow/overview.mdx:107]: I'm not sure about this whole section, a lot of very generic bullets and words here. I feel like it should be part of the "Building workflows" section, especially when you already have a usecases section at the bottom

File: workflow/overview.mdx [hunk 7]
```diff
@@ -0,0 +1,176 @@
+---
+title: "Workflow Builder Overview"
+sidebarTitle: "Overview"
+---
+
+Our browser-based **Workflow Builder (Editor)** makes it simple to create personalized, automated messaging workflows—no coding required.
```
Comment (scopsy) [inline workflow/overview.mdx:6]: Let's choose a single name here either editor or builder. Also no need to specify it's browser based

Comment (scopsy) [reply workflow/overview.mdx:6]: also, workflow editor is used to modify code based workflows. So this statement is in accurate.

File: workflow/overview.mdx [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+
+## Workflow Elements / Building Blocks
+
+All Novu workflows consist of four basic elements:
+
+- **Workflow Identifier** which is used as the trigger id 
+- A **trigger** that starts the workflow
+- **Channel Step** that send notifications to your configured channels
+- **Action Steps** that control the flow of the workflow and produce state for use in templates
```
Comment (scopsy) [inline workflow/overview.mdx:15]: Not sure I understand this one here, maybe rephrase it? What does "state for use in templates" means?

File: workflow/overview.mdx [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+## Workflow Elements / Building Blocks
+
+All Novu workflows consist of four basic elements:
+
+- **Workflow Identifier** which is used as the trigger id 
+- A **trigger** that starts the workflow
+- **Channel Step** that send notifications to your configured channels
+- **Action Steps** that control the flow of the workflow and produce state for use in templates 
+
+### Workflow Identifier (Settings)
```
Comment (scopsy) [inline workflow/overview.mdx:17]: Why the "Settings" here is needed?

File: workflow/overview.mdx [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+
+Give your Workflow a clear, descriptive name that your team can easily identify. The workflow identifier "new-signup" is used as the trigger id. Workflow identifiers should be unique to your application and descriptive of the workflow's purpose.
```
Comment (scopsy) [inline workflow/overview.mdx:19]: Where did the "new-signup" came from?

File: workflow/overview.mdx [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+For example:
+
+- Abandoned Cart Recovery
+- Password Reset Flow
+- New User Welcome Series
+- Re-engagement Campaign
+
+### Trigger
+
+Every workflow starts with a **trigger**. This is the event that starts a workflow. It represents an action or event that initiates the workflow by making a call to the Novu API, along with a specified workflow trigger identifier (`workflow_id`). The identifier helps match the trigger to the corresponding workflow.
```
Comment (scopsy) [inline workflow/overview.mdx:30]: The first 2 sentences feel almost identical, maybe rephrase it?

File: workflow/overview.mdx [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+- Abandoned Cart Recovery
+- Password Reset Flow
+- New User Welcome Series
+- Re-engagement Campaign
+
+### Trigger
+
+Every workflow starts with a **trigger**. This is the event that starts a workflow. It represents an action or event that initiates the workflow by making a call to the Novu API, along with a specified workflow trigger identifier (`workflow_id`). The identifier helps match the trigger to the corresponding workflow. 
+
+When a trigger occurs, it sends the necessary payload data that the workflow will use to deliver notifications to subscribers. **Learn more about subscribers**.
```
Comment (scopsy) [inline workflow/overview.mdx:32]: This should be a link

File: workflow/overview.mdx [hunk 13]
```diff
@@ -0,0 +1,176 @@
...
+
+Triggers are the primary and only method for sending notifications. The trigger identifier ensures that the correct workflow is executed.
+
+**Example of a Trigger**
+Here's an example of an API call to trigger an event:
+
+```bash
+curl -X POST https://api.novu.co/v2/events/trigger \
```
Comment (scopsy) [inline workflow/overview.mdx:40]: There is no v2 events endpoint to my knowledge

File: workflow/overview.mdx [hunk 14]
```diff
@@ -0,0 +1,176 @@
...
+  -H "Authorization: ApiKey <NOVU_SECRET_KEY>" \
+  -H "Content-Type: application/json" \
+  -d '{
+    "name": "<WORKFLOW_TRIGGER_IDENTIFIER>",
+    "to": {
+      "subscriberId": "<UNIQUE_SUBSCRIBER_IDENTIFIER>",
+    },
+    "payload": {}
+    }
```
Comment (scopsy) [inline workflow/overview.mdx:49]: I think this one is extra "}"

File: workflow/step-conditions.mdx [hunk 15]
```diff
@@ -0,0 +1,45 @@
+---
+title: "Step Conditions (Coming Soon)"
+sidebarTitle: "Step Conditions"
+---
+
+**Step Conditions** allow you to customize a **Subscriber's** workflow journey using their **Data**, **Topics**, **Segment Membership**, or prior behavior.
```
Comment (scopsy) [inline workflow/step-conditions.mdx:6]: What a segment membership is? we don't have any concept like this

File: mint.json (hunk 3 above)
Comment (iampearceman) [reply mint.json:142]: Done.

File: mint.json (hunk 1 above)
Comment (iampearceman) [reply mint.json:143]: Resolved.

File: workflow/overview.mdx (hunk 4 above)
Comment (iampearceman) [reply workflow/overview.mdx:8]: Addressed.

File: workflow/overview.mdx (hunk 14 above)
Comment (iampearceman) [reply workflow/overview.mdx:49]: Addressed.

File: workflow/overview.mdx (hunk 12 above)
Comment (iampearceman) [reply workflow/overview.mdx:32]: Addressed.

File: workflow/overview.mdx (hunk 11 above)
Comment (iampearceman) [reply workflow/overview.mdx:30]: Addressed.

File: workflow/overview.mdx (hunk 9 above)
Comment (iampearceman) [reply workflow/overview.mdx:17]: Addressed (removed).

File: workflow/overview.mdx (hunk 10 above)
Comment (iampearceman) [reply workflow/overview.mdx:19]: Addressed.

File: workflow/overview.mdx (hunk 8 above)
Comment (iampearceman) [reply workflow/overview.mdx:15]: Addressed.

File: workflow/overview.mdx (hunk 7 above)
Comment (iampearceman) [reply workflow/overview.mdx:6]: Addressed.

File: workflow/step-conditions.mdx (hunk 15 above)
Comment (iampearceman) [reply workflow/step-conditions.mdx:6]: Addressed.
